File: Medium/0003 - longestsubwithoutrepeatingchars/longestsubstringwithoutrepeatingcharacters.java
class Solution {
    public int lengthOfLongestSubstring(String s)
    {
        int length = 0;
        int start = 0;

        Map<Character, Integer>   mySet = new HashMap<>();
        for (int i = 0; i < s.length(); i++)
        {
            if (mySet.containsKey(s.charAt(i)))
                start = Math.max(start, mySet.get(s.charAt(i)) + 1);
            
            mySet.put(s.charAt(i), i);
            length = Math.max(length, i - start + 1);
        }
        return length;
    }
}

File: Medium/0003 - longestsubwithoutrepeatingchars/longestsubstringwithoutrepeatingcharacters.js
var lengthOfLongestSubstring = function(s)
{
    let length = 0;
    let start = 0;

    const mySet = new Map();
    for (let i = 0; i < s.length; i++)
    {
        if (mySet.has(s[i]))
            start = Math.max(start, mySet.get(s[i]) + 1);
        
        mySet.set(s[i], i);
        length = Math.max(length, i - start + 1);
    }
    return length;
};